src/datamodel/builder/Uuid.cpp

```diff
@@ -10,3 +10,99 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "up-cpp/datamodel/builder/Uuid.h"
+
+#include <random>
+#include <stdexcept>
+
+namespace uprotocol::datamodel::builder {
+
+struct UuidBuilder::UuidSharedState {
+	std::mt19937_64 random_engine{std::random_device{}()};
+	uint16_t counter{0};
+	uint64_t last_unix_ts_ms{0};
```

Comment:
minor: could this be a `std::chrono::time_point<std::chrono::system_clock, std::chrono::milliseconds>`? That way all the time comparisons and math are on chrono objects and the timestamp is only converted to an integer when it is ready to be inserted into the UUID

Reply:
Thanks, I've updated this